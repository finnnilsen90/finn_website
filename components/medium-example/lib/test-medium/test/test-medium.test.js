const React = require('react');
const ReactTestUtils = require('react-dom/test-utils');
const renderer = require('react-test-renderer')
let { shallow, mount, render } = require('enzyme');

const Button = require('../test-medium.js');

let newTest = new Button();

describe('component connection', () => {

    it('should be defined', () => {
        expect(Button).toBeDefined();
    })

})  
    
describe('<Test />', () => {

    const wrapper = render(<Button />);

    it('renders components', () => {
        expect(wrapper).toMatchSnapshot();
    });


})

describe('<Test /> State', () => {

    // using enzyme
    it('should toggle second buttons disabled state when clicking on first button', () => {
        const wrapper = shallow(<Button />);
        const firstButton = wrapper.find('button').at(0);
        const secondButton = wrapper.find('button').at(1);
    
        firstButton.simulate('click');
        expect(wrapper.state().disabled).toEqual(true);

        console.log(wrapper.state().disabled)
    
        firstButton.simulate('click');
        expect(wrapper.state().disabled).toEqual(false);
    });


})


