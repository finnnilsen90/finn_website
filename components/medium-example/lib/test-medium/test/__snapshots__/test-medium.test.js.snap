// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Test /> renders components 1`] = `
<div>
  <button
    type="button"
  >
    First Button
  </button>
  <button
    type="button"
  >
    Second Button
  </button>
</div>
`;
